adafruit_wiznet5k/adafruit_wiznet5k_socket.py

```diff
@@ -181,6 +181,13 @@ def bind(self, address):
             if ip_address != current_ip:
                 _the_interface.ifconfig = (ip_address, subnet_mask, gw_addr, dns)
         self._listen_port = address[1]
+        # For UDP servers we need to open the socket here because we won't call
+        # listen
+        if self._sock_type == SOCK_DGRAM:
+            _the_interface.socket_listen(
+                self.socknum, self._listen_port, wiznet5k.adafruit_wiznet5k.SNMR_UDP
+            )
+            self._buffer = b""
```

Comment:
What's the reason for clearing the buffer?

Reply:
We do it for the TCP `listen()` after calling `socket_listen()` below. So I am assuming we want to do it here too.